### PR TITLE
Conversion to Sessions endpoint

### PR DESCRIPTION
**Pull request by jlengrand**

### CODE REVIEW

**Comment by gcatanese** [comment]
Looks good, some comments:

- I would comment wherever `test` is used (ie link to https://checkoutshopper-test.adyen.com, et..) to make it clear this only applies when developing the integration (ie // change to live for production)
- `ApiController.GetPaymentMethods` can be removed
- Webhook missing

**Comment by abhi04-github** [comment]
I am trying to use session concept but i fail to create checkout object as i am trying not use "await" as per document. Any suggestion?

File: wwwroot/js/adyenImplementation.js [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
+const sessionId = urlParams.get('sessionId'); // Unique identifier for the payment session
+const redirectResult = urlParams.get('redirectResult');
+
+// Typical checkout experience
+async function startCheckout() {
+  // Used in the demo to know which type of checkout was chosen
+  const type = document.getElementById("type").innerHTML;
+
+  try {
+    const checkoutSessionResponse = await callServer("/api/sessions?type=" + type);
```
Comment (jonekdahl) [inline wwwroot/js/adyenImplementation.js:14]: Was the `type` parameter supposed to be used for something when the session is created? It looks like it is ignored by the API controller.

File: Controllers/ApiController.cs [hunk 2]
```diff
@@ -43,68 +43,24 @@ public ActionResult<string> GetPaymentMethods([FromBody] PaymentMethodsRequest r
...
 
-        [HttpPost("api/initiatePayment")]
-        public ActionResult<string> InitiatePayment([FromBody] PaymentRequest pmreq)
+        [HttpPost("api/sessions")]
+        public ActionResult<string> Sessions()
         {
-            _logger.LogInformation($"Request for Payments API::\n{pmreq}\n");
+            var sessionsRequest = new CreateCheckoutSessionRequest();
+            sessionsRequest.merchantAccount = _merchant_account; // required
+            sessionsRequest.channel = (CreateCheckoutSessionRequest.ChannelEnum?) PaymentRequest.ChannelEnum.Web; // required
```
Comment (jonekdahl) [inline Controllers/ApiController.cs:51]: Is `channel` really required?

File: wwwroot/js/adyenImplementation.js [hunk 3]
```diff
@@ -25,52 +53,27 @@ async function initCheckout() {
...
+          amount: {
+            value: 1000,
+            currency: "USD",
+          },
+          environment: "test", // Change this to "live" when you're ready to accept live PayPal payments
+          countryCode: "US", // Only needed for test. This will be automatically retrieved when you are in production.
         }
       },
-      onAdditionalDetails: (state, component) => {
-        handleSubmission(state, component, "/api/submitAdditionalDetails");
```
Comment (jonekdahl) [inline wwwroot/js/adyenImplementation.js:35]: Isn't this still required to [confirm additional actions](https://docs.adyen.com/online-payments/web-components/advanced-use-cases/additional-actions)?

File: wwwroot/js/adyenImplementation.js (hunk 1 above)
Comment (jlengrand) [reply wwwroot/js/adyenImplementation.js:14]: This is a remnant from before sessions where type was being used for some cases, but indeed as it is now it is not used at all. I'll remove it

File: Controllers/ApiController.cs (hunk 2 above)
Comment (jlengrand) [reply Controllers/ApiController.cs:29]: It is not, good catch. [Reference](https://docs.adyen.com/api-explorer/#/CheckoutService/v68/post/sessions__reqParam_channel)

**Comment by jlengrand** [comment]
Closes #8

File: wwwroot/js/adyenImplementation.js (hunk 3 above)
Comment (jonekdahl) [reply wwwroot/js/adyenImplementation.js:35]: It was my understanding that it was required in some cases such as QR codes for Swish, but perhaps it is optional.

Comment (jlengrand) [reply wwwroot/js/adyenImplementation.js:35]: oh, it is definitely still there! We have not removed any functionality, but it is implemented differently in some cases. You can see the call to `submitDetails` in adyenImplemention.js

**Pull request merged**